gpiozero/output_devices.py

```diff
@@ -1245,16 +1261,20 @@ def forward(self, speed=1):
 
         :param float speed:
             The speed at which the motor should turn. Can be any value between
-            0 (stopped) and the default 1 (maximum speed) if *pwm* was
-            :data:`True` when the class was constructed (and only 0 or 1 if
-            not).
+            0 (stopped) and the default 1 (maximum speed) if *pwm* or *enable_is_pwm*
+            was :data:`True` when the class was constructed (and only 0 or 1 if not).
         """
         if not 0 <= speed <= 1:
             raise ValueError('forward speed must be between 0 and 1')
-        if isinstance(self.forward_device, DigitalOutputDevice):
+        if isinstance(self.forward_device, DigitalOutputDevice) and isinstance(self.enable_device, DigitalOutputDevice):
             if speed not in (0, 1):
                 raise ValueError(
                     'forward speed must be 0 or 1 with non-PWM Motors')
+        if isinstance(self.enable_device, PWMOutputDevice):
+            self.backward_device.off()
+            self.enable_device.value = speed
+            self.forward_device.on()
+            return
         self.backward_device.off()
         self.forward_device.value = speed
```

Comment:
Again, minor nitpick but I wonder if
```python3
        if isinstance(self.enable_device, PWMOutputDevice):
            self.backward_device.off()
            self.enable_device.value = speed
            self.forward_device.on()
            return
        self.backward_device.off()
        self.forward_device.value = speed
```
would be any clearer if written as
```python3
        self.backward_device.off()
        if isinstance(self.enable_device, PWMOutputDevice):
            self.enable_device.value = speed
            self.forward_device.on()
        else:
            self.forward_device.value = speed
```
? :shrug: